src/main/java/swinglib/ErrorMessageBuilder.java

```diff
@@ -0,0 +1,123 @@
+package swinglib;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import java.awt.GraphicsEnvironment;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JOptionPane;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+
+/**
+ * Swing builder for an error message pop-up (Uses JOptionPane). Example usages:
+ * <code><pre>
+ *   JOptionPaneBuilder.builder()
+ *     .title("Something Crashed")
+ *     .message("Error Details")
+ *     .option("ok", () -> {})
+ *     .option("quit", () -> System.exit(0))
+ *     .buildAndShow();
+ * </pre></code>
+ * Default error dialog with an 'ok' button:
+ * <code><pre>
+ *   JOptionPaneBuilder.builder()
+ *     .message("error details")
+ *     .buildAndShow();
+ * </pre></code>
+ */
+public class ErrorMessageBuilder {
+
+  private String title = "Error";
+  private String message;
+  private List<CloseButton> actions = new ArrayList<>();
+
+  private static class CloseButton {
+    private final String buttonText;
+    private final Runnable closeAction;
+
+    private CloseButton(String buttonText, Runnable closeAction) {
+      this.buttonText = buttonText;
+      this.closeAction = closeAction;
+    }
+  }
+
+  private ErrorMessageBuilder() {}
+
+  public static ErrorMessageBuilder builder() {
+    return new ErrorMessageBuilder();
+  }
+
+  public ErrorMessageBuilder message(String message) {
+    this.message = message;
+    return this;
+  }
+
+  /**
+   * Constructs a Swing JDialog using current builder values.
+   * Values that must be set: title, contents
+   */
+  public void buildAndShow() {
+    checkState(!GraphicsEnvironment.isHeadless(), "JOptionPane throws an exception in headless environments");
+    checkNotNull(message);
+
+    if (actions.isEmpty()) {
+      actions.add(new CloseButton("OK", () -> {
+      }));
+    }
+
+    // First option element to JOptionPane is the right most button, right to left ordering.
+    // It is more natural for left-to-right ordering of options to be specified, so we
+    // will reverse the option ordering to allow clients to specify left to right ordering.
+
+    List<CloseButton> swingOptionsRightToLeft = reverse(actions);
```

Comment:
Running Clean Up on this file will add several missing `final` modifiers.

Reply:
Fixed in: 571b4eb70